core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
+        $field   = $query[2];
+        $search  = $query[3];
+        
+        $class_key = $elements[$element]['class_key'];
+        $fields = $elements[$element]['fields'];
+        
+        /*
+        if($element == 'modx' AND $field == 'config'){
+            // 2do lookup system settings
+        }
+        else {
+            
+        }
+        */
+        
+        // Prepare query
+        $c = $this->modx->newQuery($class_key);
+        
+        // Some extra adjustments to search for users
+        if($class_key == 'modUser'){
+            if($field != 'username'){
+                $field = 'Profile.'.$field;
+            }                
+            $c->select(array(
+                $this->modx->getSelectColumns('modUser','modUser'),
+                $this->modx->getSelectColumns('modUserProfile', 'Profile', ''),
+            ));
+            $c->leftJoin('modUserProfile', 'Profile');                    
+        }
+        elseif($class_key == 'modPlugin'){
+            
+            // 2do find plugins by event
+            // :p:event:onDocFormChange
+        }
 
+        
+        // Replace wildcards (*) with % if they are not preceeded by [ (2do: or `)
+        $search = preg_replace( '/(?<!\[)\*/', '%', $search);
+        
+        //$search = str_replace('*','%',$search);
+        
+
+        
+        $where = array();
+        
+        /* Search inside a field */
+        if(!empty($field) AND $field != '*'){
+            $like = ($search != $query[3]) ? ':LIKE' : ':=';
+
+            $where = array( $field.$like => $search );
+            $c->where($where);
+        }
+        /* Search inside all fields */
+        else {
+            // 2do I don't get why := for the else clause doesn't work...
+            $like = ($search != $query[3]) ? ':LIKE' : ':LIKE';            
+
+            foreach($fields as $fieldname){
+                $n++;
+                if($class_key == 'modUser' AND $fieldname != 'username'){
+                    if($n == 1)
+                    {
+                        $key = 'Profile.'.$fieldname.$like;                     
+                    }
+                    else
+                    {
+                        $key = 'OR:.Profile'.$fieldname.$like;
+                    }
+                }
+                else {
+                    if($n == 1)
+                    {
+                        $key = $fieldname.$like;
+                        
+                    }
+                    else
+                    {
+                        $key = 'OR:'.$fieldname.$like;
+                    }
+                }
+                $where[$key] = $search;
+            }
+            $c->where($where);
+        }            
+        
+        if($debug === true)
+        {
+            $debug['q'] = '<br/>Full query: <pre style="color: #fff">' . print_r($where,true) . '</pre>';
+        }
+        
+        $totalResults = count($this->modx->getCollection($class_key, $c));
```

Comment:
This isn't a very efficient way to check the total results, loading everything into memory ;)

Doing `$this->modx->getCount($class_key, $c)` should be better.